test/object-store/frozen_objects.cpp

```diff
@@ -289,6 +303,32 @@ TEST_CASE("Freeze Results", "[freeze_results]") {
         realm->close();
         REQUIRE(DB::call_with_lock(config.path, [](auto) {}));
     }
+
+    SECTION("Results after owning object remove") {
+        Obj obj;
+        write([&]() {
+            obj = create_object(42);
+        });
+
+        Results results(realm, obj.get_dictionary_ptr(int_dict_col));
+        REQUIRE(results.is_valid());
+        REQUIRE(results.size() == 5);
+
+        write([&]() {
+            obj.remove();
+        });
+
+        REQUIRE_EXCEPTION(results.size(), StaleAccessor, "Access to invalidated Results objects");
+        REQUIRE_EXCEPTION(results.get_any(0).is_null(), StaleAccessor, "Access to invalidated Results objects");
+        REQUIRE_FALSE(results.is_valid());
+
+        Results frozen_res = results.freeze(realm);
+
+        // FIXME would probably be better to throw an exception?
```

Comment:
I guess we can't really throw any exception unless we don't make a breaking change.

Reply:
I've put this on the wrong line, it's not obvious. Currently without the fix 'freeze' actually crashing, so "throw an exception" is meant for it, so that there is no ambiguity with what properties frozen results should have after freeze.